src/Compilers/CSharp/Portable/Binder/Binder_Expressions.cs

```diff
@@ -8455,7 +8455,6 @@ private static CSharpSyntaxNode GetConditionalReceiverSyntax(ConditionalAccessEx
         private BoundExpression GetReceiverForConditionalBinding(ExpressionSyntax binding, DiagnosticBag diagnostics)
         {
             var conditionalAccessNode = SyntaxFactory.FindConditionalAccessNodeForBinding(binding);
-            Debug.Assert(conditionalAccessNode != null);
```

Comment:
>Debug.Assert(conditionalAccessNode != null); [](start = 12, length = 44)

If assert is removed, how a reader would know that it is not null? Consider providing explicit type for the variable.